pandas/io/formats/csvs.py

```diff
@@ -150,6 +150,16 @@ def save(self):
 
             self._save()
 
+            # GH 17778 handles compression for byte strings.
```

Comment:
why are you handling this here and not in the finally?

Reply:
moved to finally.